### PR TITLE
Adds the Vector (SMG)

### PR DESCRIPTION
**Pull request by DrSnips**
Adds a gun that matches the intent of the real version. The idea is that it allows for you to use nearly any type of mag/ammo you have on hand. The receiver part of the gun comes off by applying a screwdriver to it while the mag is removed. Once removed you can apply a mag to the receiver which will then "adapt" to the applied mag/ammo. Then you can reapply the receiver to the gun to continue using your selected ammo type.
I will update the wiki after any requested changes.
Other notes:
It's an SMG so there is a burst fire but its only two rounds rather than three.
It only can change to ammo that goes in magazines.
Changing the receiver requires you to stand still for 10 seconds.
vector sprites

:cl:
* rscadd: Added the Vector. You can remove the receiver via screwdriver and then change its ammo/mag type by applying a magazine to the receiver. It starts with standard glock ammo (.380auto).

### CODE REVIEW

**Comment by DrSnips** [comment]
If icondiff bot is dead I will post the sprites later.

**Comment by Pathid** [comment]
The bot reports go into a tab now.

**Comment by Blithering** [comment]
This and the goonecode ammo stockpile are gonna make a dangerous combination. Icondiffbot says you deleted the empty stechkin pistol sprites though

**Comment by DrSnips** [comment]
That was one if the hopes. But I realized that most of that ammo is just boxes and no mags so it can't be used. Another PR at another time I guess. .75 ammo is neat since you can punch through walls kinda fast. .666 is revolver size and doesn't use mags so it can't be used but again thats another PR.

**Comment by Dacendeth** [comment]
What happens if you try to use a lawgiver magazine on it?

**Comment by Blithering** [comment]
good point about the ammo boxes. Guess the major silliness will be limited to deff maint spawns then.

**Comment by DrSnips** [comment]
@Dacendeth It rejects lawgiver mags because they aren't like normal ones see line 345. If you can think of any other mags that don't obey normal mag rules please say so.

**Comment by DrSnips** [comment]
I was thinking of putting one or two of these in the sec armory since they are technically just fancy glocks to start. Anyone have any other ideas?

**Comment by Blithering** [comment]
Maybe the vox dart gun cartridges count as magazines? They'd be pretty wonky if they worked in the vector.

**Comment by DrSnips** [comment]
OP updated with sprites in case anyone missed them
@Blithering The vox darts aren't a child of magazines so it's not accepted. They are a snowflake item.

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 1]
```diff
@@ -225,6 +225,142 @@
...
+		icon_state = "vector_assembly"
+		item_state = "vector_assembly"
+		name = "\improper Vector Assembly"
+		desc = "A lightweight and compact gun, it's receiver has been removed."
+	if(istype(loc, /mob))
+		var/mob/M = loc
+		M.update_inv_hands()
+
+/obj/item/weapon/gun/projectile/automatic/vector/can_discharge()
+	..()
```
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/automatic.dm]: .=..() so you inherit the return value of the parent call

**Comment by Blithering** [comment]
That max shells 25 thing reminded me, does the LMG box magazine work in this?

**Comment by DrSnips** [comment]
If its a magazine and not the lawgiver's it works. Some suspension of disbelief is required. If that's the saw one yes it works. Im not too worried about it since that admin only iirc so if the admins are dropping a saw in your lap its past the point of balance issues.

**Comment by DrSnips** [comment]
Mobile posting was a mistake.

**Comment by DrSnips** [comment]
This PR is done. I will map them in later because my first try resulted in a lot of mustard gas being made with map conflicts and git related issues.

File: code/modules/projectiles/guns/projectile/automatic.dm (hunk 1 above)
Comment (DrSnips) [reply code/modules/projectiles/guns/projectile/automatic.dm]: Git is being retarded and not hiding this after its been **fixed**.

**Pull request merged**